synapse/lib/storm.py

```diff
@@ -488,18 +501,56 @@ def cmpr(tufo):
 
     def _cmprCtorTag(self, oper):
         tag = self._reqOperArg(oper,'valu')
-
-        props = {}
-        def cmpr(tufo):
+        glob_props = collections.defaultdict(set)
+        reg_props = {}
+
+        if tag.startswith(self._rt_glob_smmark) or self._rt_glob_smark in tag or self._rt_glob_mmark in tag:
+            # Prep the tag regex
+            _tag = tag.lower()
+            if _tag.startswith(self._rt_glob_smmark):
+                _tag = _tag.replace(self._rt_glob_smmark, self._rt_glob_smsentinal, 1)
+            while self._rt_glob_smark in _tag:
+                _tag = _tag.replace(self._rt_glob_smark, self._rt_glob_ssentinal)
+            while self._rt_glob_mmark in _tag:
```

Comment:
Actually, splitting the above tag into a list ( from the above comment ) allows these to be a list comprehension rather than string replacement loops.
```
parts = [ p for p in parts if p != '**' else globre ]
parts = [ p for p in parts if p != '*' else otherre ]
_tag = '.'.join(parts)
```